test/e2e/helmreleaseproxy_helpers.go

```diff
@@ -0,0 +1,88 @@
+//go:build e2e
+
+// Copyright 2024 D2iQ, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/gomega"
+	addonsv1 "sigs.k8s.io/cluster-api-addon-provider-helm/api/v1alpha1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/test/framework"
+	"sigs.k8s.io/cluster-api/util/conditions"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// WaitForHelmReleaseProxyReadyInput is the input for WaitForHelmReleaseProxyReady.
+type WaitForHelmReleaseProxyReadyForClusterInput struct {
+	GetLister          framework.GetLister
+	Cluster            *clusterv1.Cluster
+	HelmChartProxyName string
+}
+
+// WaitForHelmReleaseProxyReady waits until the HelmReleaseProxy has ready condition = True, that signals that the Helm
+// install was successful.
+func WaitForHelmReleaseProxyReadyForCluster(
```

Comment:
This kind of already exists in CAAPH but pulling that in would require CAAPH as a code dependency which we have been trying to avoid by importing external APIs and I don't think it's worth changing this for this one function.